generate.mjs

```diff
@@ -0,0 +1,101 @@
+
+import {spawnSync} from 'node:child_process';
+import fs from 'node:fs';
+import path from 'node:path';
+
+function execute(cmd, args, options) {
+  const { error, status } = spawnSync(cmd, args, {...options || {}, shell: true, stdio: 'inherit'});
+  if (error) {
+    throw new(Error);
+  }
+  if (status !== 0) {
+    throw new Error(`${cmd} exited with status code: ${status}`);
+  }
+}
+
+function readFileAsUTF8(filename) {
+  return fs.readFileSync(filename, { encoding: 'utf-8' });
+}
+
+function insertIncludes(filename) {
+  const dir = path.dirname(filename);
+  const bs = readFileAsUTF8(filename)
+  
+    .replace(/<pre\s+class=include\s*>\s*path:\s*(.*)\s*<\/pre>/gm, (m, filename) => {
+      return readFileAsUTF8(path.join(dir, filename));
+    });
+
+  fs.writeFileSync(filename, bs);
```

Comment:
Oh, the problem is that this modifies `gpuweb/spec/index.bs` in place, making the submodule dirty.

Reply:
I think I don't understand something there. The instructions are to edit that file manually by copying copies.bs into it so that's the difference if you do it by hand or this script does it? Both modify it and should "make it dirty"

Reply:
Oh, good question. I forgot we fixed this but those instructions are outdated, that step isn't needed anymore: #172

That explains how the bot was passing (producing a correct generated output) before.

Reply:
Well, I fixed it, and I didn't bother to get anyone to review it ^_^
Sorry I forgot to update the readme.

Reply:
Thanks!